sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -329,13 +351,15 @@ def _set_partition_key(event_datas, partition_key):
         self._check_closed()
         if isinstance(event_data, EventData):
             if partition_key:
-                event_data._set_partition_key(partition_key)
+                event_data._set_partition_key(partition_key)  # pylint: disable=protected-access
             wrapper_event_data = event_data
         else:
-            event_data_with_pk = self._set_partition_key(event_data, partition_key)
-            wrapper_event_data = _BatchSendEventData(
-                event_data_with_pk,
-                partition_key=partition_key) if partition_key else _BatchSendEventData(event_data)
+            if isinstance(event_data, EventDataBatch):
+                wrapper_event_data = event_data
```

Comment:
Does this mean that in the case of EventDataBatch we ignore the passed in partition key?

Reply:
Yes, EventDataBatch wins. It also has partition_key.

Reply:
In that case we should raise an error if a user tries to pass in the partition_key alongside a Batch object. Otherwise they might think the value they pass in being used when it's actually being ignored.